salt/states/pkg.py

```diff
@@ -165,6 +165,10 @@ def _fulfills_version_spec(versions, oper, desired_version,
     otherwise returns False
     '''
     cmp_func = __salt__.get('pkg.version_cmp')
+    # stripping "with_origin" dict wrapper
+    if salt.utils.is_freebsd():
+        if type(versions) is dict and 'version' in versions:
```

Comment:
Type checking should be using isinstance, since it more gracefully handles subclasses. Not saying we'll ever subclass dict here, but it's a good way of future-proofing against potential future breakage.

With regard to disabling ``with_origin``, I'm not familiar with this option, I did not add support for it. It seems to default to ``False`` though, looking at the source code.